src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyDetectorJobRequest.java

```diff
@@ -33,24 +33,45 @@
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
 
+import com.amazon.opendistroforelasticsearch.ad.model.DetectionDateRange;
+
 public class AnomalyDetectorJobRequest extends ActionRequest {
 
     private String detectorID;
+    private DetectionDateRange detectionDateRange;
+    private boolean historical;
     private long seqNo;
     private long primaryTerm;
     private String rawPath;
 
     public AnomalyDetectorJobRequest(StreamInput in) throws IOException {
         super(in);
         detectorID = in.readString();
+        if (in.readBoolean()) {
+            detectionDateRange = new DetectionDateRange(in);
+        }
+        historical = in.readBoolean();
```

Comment:
I think we should have some code logic to make sure the input has valid detection data range if the historical is true, right?

Reply:
`DetectionDateRange` has validation logic: https://github.com/opensearch-project/anomaly-detection/blob/single-flow-development/src/main/java/com/amazon/opendistroforelasticsearch/ad/model/DetectionDateRange.java#L66

Here the `detectionDateRange` and `historical` boolean can be passed in individually. The `historical` flag is for stopping detector, the `detectionDateRange` is for starting detector. So it's ok if `historical is true` but `detectionDateRange is null`